engine/src/main/java/org/terasology/entitySystem/ComponentContainer.java

```diff
@@ -25,6 +27,13 @@
      */
     boolean hasComponent(Class<? extends Component> component);
 
+    /**
+     * @param filterComponents
+     * @return If this has at least one component from the list of components
+     */
+    boolean hasComponents(List<Class<? extends Component>> filterComponents);
```

Comment:
I did a quick API impact check in my mega-workspace, looks like there are no other usages (for extends anyway - there's a grand total of one class usage in GooeyDefence) so we might be OK without a default :-)

Pity we end up with several identical implementations yet we can't really stick them in here since they search a variable not yet available.

However, I would have two suggestions here:

1. Should we rename this to `hasAnyComponents` to indicate the nature of the check? Then maybe also add `hasAllComponents` while we're at it?
1. Slightly more javadoc would be nice, with a line for the main description and a few words on the param+return :-)

Reply:
I agree with @Cervator's suggestions, especially the renaming.

Reply:
Done.

Reply:
👍